headers/xdp/xsk.h

```diff
@@ -258,11 +258,13 @@ struct xsk_socket_config {
  * The following fields are optional:
  * 
  * @fill, @comp, @rx_size, @tx_size, @libxdp_flags, @xdp_flags,
- * @bind_flags
+ * @bind_flags, @shared
  *  If a socket with exclusive ownership of a umem is going to be
- *  created, keep @fill and @comp unset. If the umem is to be shared
- *  with other sockets, set @fill and @comp to the corresponding
- *  fields of the umem.
+ *  created, leave @shared unset as its default value is false.
+ *  Values of @fill and @comp will be ignored if @shared is unset
+ *  or set to false. 
+ *  If the umem is to be shared with other sockets, set @shared to 
+ *  true, @fill and @comp to the corresponding fields of the umem.
```

Comment:
Erm, why is an extra field needed? If the intention is that `fill` and `comp` are used only if shared is set, and ignored otherwise, the `shared` field provides no extra information: simply setting `fill` and `comp` (or, conversely, setting them both to NULL) should be enough to signal the intention of the caller?

Reply:
> should be enough to signal the intention of the caller
I think you are right